.github/workflows/build.yml

```diff
@@ -37,19 +37,63 @@ jobs:
           python setup.py bdist_wheel
 
       - name: Run Tests
-        run : pipenv run tests # execute Pipfile script tests
-
-      - name: Build Docs
-        run: pipenv run docs # build docs
+        run : pipenv run tests
 
       - name: Coveralls
-        # the official action only uses Lcov files which pytest-cov does not support
-        # https://github.com/coverallsapp/github-action/issues/30
-        #
-        # using https://github.com/coveralls-clients/coveralls-python instead
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           pip install coveralls
           coveralls
 
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+          ref: master
+          # require all of history to build all tagged versions' docs
+          fetch-depth: 0
+
+      # also checkout the branch that triggered this build
+      - name: Checkout Source
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Install Dependencies
+        run: |
+          env
+          pip install pipenv
+          pipenv install --dev
+
+      - name: Checkout gh-pages
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          path: gh-pages
+
+      - name: Update gh-pages
+        run: |
+          rm -r gh-pages/*
+          cp docs/index.html gh-pages/
+
+      - name: Build Docs
+        # this builds all tagged versions and master
+        run: |
+          env
+          pipenv run sphinx-multiversion docs gh-pages/
+
+      - name: Push gh-pages
+        uses: EndBug/add-and-commit@v5
+        with:
+          message: "updating built documentation"
+          branch: gh-pages
+          add: ". --force"
+          cwd: "./gh-pages/"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
How do we tell GitHub about this token? We have dls-py3 command to push the PYPI token to Travis, is there a GitHub analogy here?

Reply:
This is GitHub's own token that it already knows about. You can add more secrets in the web UI. I'll get Pypi publishing working soon and see what is required.